### PR TITLE
Classifier ABC

### PR DESCRIPTION
**Pull request by amburger66**
- adds the Classifier class
- makes MLPClassifier inherit from it
- replaces MLPClassifier in type definitions that can be more general

this is in preparation for other types of classifiers, like an MLPClassifierEnsemble I implemented

### CODE REVIEW

File: src/torch_models.py [hunk 1]
```diff
@@ -278,7 +278,29 @@ def _normalize_data(data: Tensor) -> Tuple[Tensor, Tensor, Tensor]:
...
+
+    def fit(self, X: Array, y: Array) -> None:
+        """Train classifier on the given data.
+
+        X is multi-dimensional, y is single-dimensional.
+        """
+        raise NotImplementedError("Override me")
+
+    def forward(self, inputs: Array) -> Tensor:
+        """Pytorch forward method."""
```
Comment (tomsilver) [inline src/torch_models.py:292]: Can probably remove this method, since not classifiers need one, intuitively.

**Review by tomsilver — COMMENTED** [review]
Looking good thank you! I left some comments suggesting a more general implementation. If you think it's better to stick with PyTorch, though, another option is to rename Classifer to PytorchClassifier and keep what you have. I'll leave the judgement call to you!

**Review by tomsilver — APPROVED** [review]
Lgtm thank you!

File: src/torch_models.py [hunk 2]
```diff
@@ -278,7 +279,27 @@ def _normalize_data(data: Tensor) -> Tuple[Tensor, Tensor, Tensor]:
         return (data - shift) / scale, shift, scale
 
 
-class MLPClassifier(nn.Module):
+class Classifier(abc.ABC):
+    """ABC for classifier types."""
+
+    @abc.abstractmethod
+    def fit(self, X: Array, y: Array) -> None:
+        """Train classifier on the given data.
```
Comment (ronuchit) [inline src/torch_models.py:287]: Sorry to revive a merged PR! @amburger66 could you remove the duplicate docstrings on the fit() and classify() in MLPClassifier below? Linter won't complain bc you don't need a docstring for overriden methods, and so it's better to have fewer duplicate lines when possible.

Comment (amburger66) [reply src/torch_models.py:287]: my bad, will do!

**Pull request merged**